src/functions/org/apache/jmeter/functions/TimeShiftingFunction.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.jmeter.functions;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.Year;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.jmeter.engine.util.CompoundVariable;
+import org.apache.jmeter.samplers.SampleResult;
+import org.apache.jmeter.samplers.Sampler;
+import org.apache.jmeter.threads.JMeterVariables;
+import org.apache.jmeter.util.JMeterUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * timeShifting Function permit to shift a date
+ *
+ * Parameters: - format date @see
+ * https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html
+ * (optional - defaults to epoch time in millisecond) - date to shift formated
+ * as first param (optional - defaults now) - amount of (seconds / minutes /
+ * hours / days / months ) to add (optional - default nothing is add ) -
+ * variable name ( optional )
+ *
+ * Returns: - Returns a formated date with the specified number of (seconds /
+ * minutes / hours / days / months ) added. - value is also saved in the
+ * variable for later re-use.
+ *
+ * @since 3.3
+ */
+public class TimeShiftingFunction extends AbstractFunction {
+    private static final Logger log = LoggerFactory.getLogger(TimeShiftingFunction.class);
+
+    private static final String KEY = "__timeShifting"; // $NON-NLS-1$
```

Comment:
"timeShifting" is unusual style for function name. Usually functions are called like "__timeShift", without continuous tense